pyiron_workflow/atomistics_library/macronodes.py

```diff
@@ -0,0 +1,64 @@
+from pyiron_workflow.macro import Macro, macro_node
+from pyiron_workflow.atomistics_library.calculatornodes import calc_with_calculator
+from pyiron_workflow.atomistics_library.tasknodes import (
+    get_elastic_matrix_task_generator,
+    get_evcurve_task_generator,
+    get_phonons_task_generator,
+    analyse_structures,
+    generate_structures
+)
+
+
+@macro_node()
+def get_energy_volume_curve(wf: Macro) -> None:
+    wf.get_task_generator = get_evcurve_task_generator()
+    wf.generate_structures = generate_structures(instance=wf.get_task_generator)
+    wf.calc_with_calculator = calc_with_calculator(task_dict=wf.generate_structures)
+    wf.fit = analyse_structures(instance=wf.get_task_generator, output_dict=wf.calc_with_calculator)
```

Comment:
These three nodes are required for every workflow, so I would like to create a macro like:
```
@macro_node()
def internal(wf: Macro) -> None:
    wf.generate_structures = generate_structures(instance=wf.get_task_generator)
    wf.calc_with_calculator = calc_with_calculator(task_dict=wf.generate_structures)
    wf.fit = analyse_structures(instance=wf.get_task_generator, output_dict=wf.calc_with_calculator)
    wf.inputs_map = {
        "generate_structures__instance": "instance",
        "analyse_structures__instance": "instance",
        "calc_with_calculator__calculator": "calculator",
    }
    wf.outputs_map = {"fit__fit_dict": "fit_dict"}
```

The issue is that both `generate_structures()` and `analyse_structures()` should receive the same `instance` as input. Is this possible

Reply:
It's possible to ensure that both `generate_structures()` and `analyse_structures()` get the same `instance` input, but not by leveraging the `inputs_map`. The `inputs/outputs_map` are (except for `None` which turns IO off) bijective maps. That means enforcing the same input needs to be done with graph connections rather than with the IO mapping. We could consider changing this expectation, but having the IO map be bijective has advantages that we'd be sacrificing for this convenience.

In the mean time, you can use a `standard.UserInput` node to take the `instance` input, and then connect its output (which is just plainly passed through the node)  to both targets:

```python
@macro_node()
def internal(wf: Macro) -> None:
    wf.instance = wf.create.standard.UserInput()
    wf.generate_structures = generate_structures(instance=wf.instance)
    wf.calc_with_calculator = calc_with_calculator(task_dict=wf.generate_structures)
    wf.fit = analyse_structures(output_dict=wf.calc_with_calculator, instance=wf.instance)
    wf.inputs_map = {
        "instance__user_input": "instance",
        "calc_with_calculator__calculator": "calculator",
    }
    wf.outputs_map = {"fit__fit_dict": "fit_dict"}
```

I imagine `UserInput` being useful once we have a GUI too, but this is actually an intended use of the design: mapping user input into the macro so that a single value can go multiple places. IMO the extra one line of inconvenience is worth it in order for the purity of the bijective mapping.

The only thing I _don't_ like is that I'm not sure how easy it is to type-hint this. It might be as simple as `wf.instance.inputs.user_input.type_hint = WhateverClass; wf.instance.outputs.user_input.type_hint = WhateverClass`, but I would need to dig in and see if the type checker is really storing raw types or if they get processed (e.g. `int | float` for sure gets converted to something like `typing.Union[int, float]`). The positive side is that such post-facto modifications of the type hinting are, in principle, possible and relatively straightforward, so it's just a matter of cooking up sexy syntax for it at a later date.

Reply:
I was able to solve this by defining a simple `get_instance()` node:
```
@single_value_node("instance")
def get_instance(instance):
    return instance
```